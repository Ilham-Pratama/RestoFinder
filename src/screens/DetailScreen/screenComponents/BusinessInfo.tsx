import React from 'react';
import { useTheme } from '@react-navigation/native';
import { Linking, Pressable, StyleSheet, View } from 'react-native';
import Header from 'components/Header';
import ThemedText from 'components/ThemedText';
import DiamondTurnRightIcon from 'assets/svgs/DiamondTurnRightIcon';
import PhoneIcon from 'assets/svgs/PhoneIcon';
import ArrowUpRightFromSquareIcon from 'assets/svgs/ArrowUpRightFromSquareIcon';
import { Business } from 'types/Business';
import sharedStyles, { ON_PRESS_BACKGROUND_COLOR } from 'shared/styles';

interface BusinessInfoProps extends Business {
  onVisitYelpSite?: (value: string) => void;
}

const BusinessInfo = ({ onVisitYelpSite, ...data }: BusinessInfoProps) => {
  const theme = useTheme();

  return (
    <View
      style={[
        { backgroundColor: theme.colors.card },
        sharedStyles.paddingVertical4,
        sharedStyles.marginTop2
      ]}>
      <Header>Business Info</Header>
      <View
        style={[
          { borderBottomColor: theme.dark ? '#3a3a3a' : '#e0e0e0' },
          styles.businessInfo,
          sharedStyles.paddingVertical4,
          sharedStyles.marginHorizontal3
        ]}>
        <View>
          {data.location.address1 ? (
            <ThemedText accessibilityRole="text">
              {data.location.address1}
            </ThemedText>
          ) : null}
          {data.location.address2 ? (
            <ThemedText accessibilityRole="text">
              {data.location.address2}
            </ThemedText>
          ) : null}
          {data.location.address3 ? (
            <ThemedText accessibilityRole="text">
              {data.location.address3}
            </ThemedText>
          ) : null}
          <ThemedText accessibilityRole="text">
            {data.location.city}, {data.location.state} {data.location.zip_code}
          </ThemedText>
        </View>
        <DiamondTurnRightIcon width={20} height={20} fill={theme.colors.text} />
      </View>
      <View
        style={[
          { borderBottomColor: theme.dark ? '#3a3a3a' : '#e0e0e0' },
          styles.businessInfo,
          sharedStyles.paddingVertical4,
          sharedStyles.marginHorizontal3
        ]}>
        <ThemedText accessibilityRole="text">{data.display_phone}</ThemedText>
        <PhoneIcon width={20} height={20} fill={theme.colors.text} />
      </View>
      <Pressable
        accessibilityRole="button"
        style={[
          styles.businessInfo,
          sharedStyles.paddingVertical4,
          sharedStyles.paddingHorizontal3,
          { borderBottomWidth: 0 }
        ]}
        android_ripple={{
          color: ON_PRESS_BACKGROUND_COLOR
        }}
        onPress={() =>
          onVisitYelpSite
            ? onVisitYelpSite(data.url)
            : Linking.openURL(data.url)
        }>
        <ThemedText>Visit Yelp Site</ThemedText>
        <ArrowUpRightFromSquareIcon
          width={20}
          height={20}
          fill={theme.colors.text}
        />
      </Pressable>
    </View>
  );
};

const styles = StyleSheet.create({
  businessInfo: {
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'space-between',
    borderBottomWidth: 1
  }
});

export default BusinessInfo;
