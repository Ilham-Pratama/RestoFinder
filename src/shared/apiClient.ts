import axios from 'axios';
import { YELP_API_TOKEN } from 'react-native-dotenv';

export const BASE_API_URL = 'https://api.yelp.com/v3';

export const apiClient = axios.create({
  baseURL: BASE_API_URL,
  timeout: 1000,
  headers: {
    Authorization: `Bearer ${YELP_API_TOKEN}`
  }
});
