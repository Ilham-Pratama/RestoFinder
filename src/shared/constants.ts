import { ICategoryItem } from 'components/CategoryItem/types';

export const MENU_CATEGORIES: ICategoryItem[] = [
  {
    label: 'Burger',
    value: 'burgers',
    image: require('assets/images/burger.png')
  },
  {
    label: 'Dessert',
    value: 'desserts',
    image: require('assets/images/cake.png')
  },
  {
    label: 'Pasta',
    value: 'pastashops',
    image: require('assets/images/pasta.png')
  },
  {
    label: 'Pizza',
    value: 'pizza',
    image: require('assets/images/pizza.png')
  },
  {
    label: 'Smoothies',
    value: 'juicebars',
    image: require('assets/images/smoothies.png')
  },
  {
    label: 'steak',
    value: 'steak',
    image: require('assets/images/steak.png')
  }
];
