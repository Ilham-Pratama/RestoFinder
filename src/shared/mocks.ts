export const BUSINESS_DUMMY = {
  id: 'WavvLdfdP6g8aZTtbBQHTw',
  alias: 'gary-danko-san-francisco',
  name: 'Gary Danko',
  image_url:
    'https://s3-media2.fl.yelpcdn.com/bphoto/CPc91bGzKBe95aM5edjhhQ/o.jpg',
  is_claimed: true,
  is_closed: false,
  url: 'https://www.yelp.com/biz/gary-danko-san-francisco?adjust_creative=wpr6gw4FnptTrk1CeT8POg&utm_campaign=yelp_api_v3&utm_medium=api_v3_business_lookup&utm_source=wpr6gw4FnptTrk1CeT8POg',
  phone: '+14157492060',
  display_phone: '(415) 749-2060',
  review_count: 5296,
  categories: [
    {
      alias: 'newamerican',
      title: 'American (New)'
    },
    {
      alias: 'french',
      title: 'French'
    },
    {
      alias: 'wine_bars',
      title: 'Wine Bars'
    }
  ],
  rating: 4.5,
  location: {
    address1: '800 N Point St',
    address2: '',
    address3: '',
    city: 'San Francisco',
    zip_code: '94109',
    country: 'US',
    state: 'CA',
    display_address: ['800 N Point St', 'San Francisco, CA 94109'],
    cross_streets: ''
  },
  coordinates: {
    latitude: 37.80587,
    longitude: -122.42058
  },
  photos: [
    'https://s3-media2.fl.yelpcdn.com/bphoto/CPc91bGzKBe95aM5edjhhQ/o.jpg',
    'https://s3-media4.fl.yelpcdn.com/bphoto/FmXn6cYO1Mm03UNO5cbOqw/o.jpg',
    'https://s3-media4.fl.yelpcdn.com/bphoto/HZVDyYaghwPl2kVbvHuHjA/o.jpg'
  ],
  price: '$$$$',
  hours: [
    {
      open: [
        {
          is_overnight: false,
          start: '1730',
          end: '2200',
          day: 0
        },
        {
          is_overnight: false,
          start: '1730',
          end: '2200',
          day: 1
        },
        {
          is_overnight: false,
          start: '1730',
          end: '2200',
          day: 2
        },
        {
          is_overnight: false,
          start: '1730',
          end: '2200',
          day: 3
        },
        {
          is_overnight: false,
          start: '1730',
          end: '2200',
          day: 4
        },
        {
          is_overnight: false,
          start: '1730',
          end: '2200',
          day: 5
        },
        {
          is_overnight: false,
          start: '1730',
          end: '2200',
          day: 6
        }
      ],
      hours_type: 'REGULAR',
      is_open_now: false
    }
  ],
  transactions: [],
  special_hours: [
    {
      date: '2019-02-07',
      is_closed: null,
      start: '1600',
      end: '2000',
      is_overnight: false
    }
  ]
};

export const REVIEW_DUMMY = {
  id: 'xAG4O7l-t1ubbwVAlPnDKg',
  rating: 5,
  user: {
    id: 'W8UK02IDdRS2GL_66fuq6w',
    profile_url:
      'https://www.yelp.com/user_details?userid=W8UK02IDdRS2GL_66fuq6w',
    image_url:
      'https://s3-media3.fl.yelpcdn.com/photo/iwoAD12zkONZxJ94ChAaMg/o.jpg',
    name: 'Ella A.'
  },
  text: 'Went back again to this place since the last time i visited the bay area 5 months ago, and nothing has changed. Still the sketchy Mission, Still the cashier...',
  time_created: '2016-08-29 00:41:13',
  url: 'https://www.yelp.com/biz/la-palma-mexicatessen-san-francisco?hrid=hp8hAJ-AnlpqxCCu7kyCWA&adjust_creative=0sidDfoTIHle5vvHEBvF0w&utm_campaign=yelp_api_v3&utm_medium=api_v3_business_reviews&utm_source=0sidDfoTIHle5vvHEBvF0w'
};

export const REVIEWS_DUMMY = {
  reviews: [
    {
      id: 'xAG4O7l-t1ubbwVAlPnDKg',
      rating: 5,
      user: {
        id: 'W8UK02IDdRS2GL_66fuq6w',
        profile_url:
          'https://www.yelp.com/user_details?userid=W8UK02IDdRS2GL_66fuq6w',
        image_url:
          'https://s3-media3.fl.yelpcdn.com/photo/iwoAD12zkONZxJ94ChAaMg/o.jpg',
        name: 'Ella A.'
      },
      text: 'Went back again to this place since the last time i visited the bay area 5 months ago, and nothing has changed. Still the sketchy Mission, Still the cashier...',
      time_created: '2016-08-29 00:41:13',
      url: 'https://www.yelp.com/biz/la-palma-mexicatessen-san-francisco?hrid=hp8hAJ-AnlpqxCCu7kyCWA&adjust_creative=0sidDfoTIHle5vvHEBvF0w&utm_campaign=yelp_api_v3&utm_medium=api_v3_business_reviews&utm_source=0sidDfoTIHle5vvHEBvF0w'
    },
    {
      id: '1JNmYjJXr9ZbsfZUAgkeXQ',
      rating: 4,
      user: {
        id: 'rk-MwIUejOj6LWFkBwZ98Q',
        profile_url:
          'https://www.yelp.com/user_details?userid=rk-MwIUejOj6LWFkBwZ98Q',
        image_url: null,
        name: 'Yanni L.'
      },
      text: 'The "restaurant" is inside a small deli so there is no sit down area. Just grab and go.\n\nInside, they sell individually packaged ingredients so that you can...',
      time_created: '2016-09-28 08:55:29',
      url: 'https://www.yelp.com/biz/la-palma-mexicatessen-san-francisco?hrid=fj87uymFDJbq0Cy5hXTHIA&adjust_creative=0sidDfoTIHle5vvHEBvF0w&utm_campaign=yelp_api_v3&utm_medium=api_v3_business_reviews&utm_source=0sidDfoTIHle5vvHEBvF0w'
    },
    {
      id: 'SIoiwwVRH6R2s2ipFfs4Ww',
      rating: 4,
      user: {
        id: 'rpOyqD_893cqmDAtJLbdog',
        profile_url:
          'https://www.yelp.com/user_details?userid=rpOyqD_893cqmDAtJLbdog',
        image_url: null,
        name: 'Suavecito M.'
      },
      text: 'Dear Mission District,\n\nI miss you and your many delicious late night food establishments and vibrant atmosphere.  I miss the way you sound and smell on a...',
      time_created: '2016-08-10 07:56:44',
      url: 'https://www.yelp.com/biz/la-palma-mexicatessen-san-francisco?hrid=m_tnQox9jqWeIrU87sN-IQ&adjust_creative=0sidDfoTIHle5vvHEBvF0w&utm_campaign=yelp_api_v3&utm_medium=api_v3_business_reviews&utm_source=0sidDfoTIHle5vvHEBvF0w'
    }
  ],
  total: 3,
  possible_languages: ['en']
};
