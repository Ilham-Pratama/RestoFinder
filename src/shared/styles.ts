import { StyleSheet } from 'react-native';

export const ON_PRESS_BACKGROUND_COLOR = 'rgba(160, 160, 160, 0.2)';

const sharedStyles = StyleSheet.create({
  elevation: {
    shadowColor: 'black',
    shadowOffset: {
      height: 5,
      width: 5
    },
    elevation: 3,
    shadowOpacity: 0.1
  },

  flex1: {
    flex: 1
  },

  flexDirectionRow: {
    flexDirection: 'row'
  },

  alignItemsCenter: {
    alignItems: 'center'
  },

  justifyContentCenter: {
    justifyContent: 'center'
  },
  justifyContentSpaceBetween: {
    justifyContent: 'space-between'
  },

  padding1: {
    padding: 5
  },
  padding2: {
    padding: 10
  },
  padding3: {
    padding: 15
  },
  padding4: {
    padding: 20
  },

  paddingHorizontal1: {
    paddingHorizontal: 5
  },
  paddingHorizontal2: {
    paddingHorizontal: 10
  },
  paddingHorizontal3: {
    paddingHorizontal: 15
  },
  paddingHorizontal4: {
    paddingHorizontal: 20
  },

  paddingVertical1: {
    paddingVertical: 5
  },
  paddingVertical2: {
    paddingVertical: 10
  },
  paddingVertical3: {
    paddingVertical: 15
  },
  paddingVertical4: {
    paddingVertical: 20
  },

  marginTop1: {
    marginTop: 5
  },
  marginTop2: {
    marginTop: 10
  },
  marginTop3: {
    marginTop: 15
  },
  marginTop4: {
    marginTop: 20
  },

  marginBottom1: {
    marginBottom: 5
  },
  marginBottom2: {
    marginBottom: 10
  },
  marginBottom3: {
    marginBottom: 15
  },
  marginBottom4: {
    marginBottom: 20
  },

  marginRight1: {
    marginRight: 5
  },
  marginRight2: {
    marginRight: 10
  },
  marginRight3: {
    marginRight: 15
  },
  marginRight4: {
    marginRight: 20
  },

  marginHorizontal1: {
    marginHorizontal: 5
  },
  marginHorizontal2: {
    marginHorizontal: 10
  },
  marginHorizontal3: {
    marginHorizontal: 15
  },
  marginHorizontal4: {
    marginHorizontal: 20
  },

  marginVertical1: {
    marginVertical: 5
  },
  marginVertical2: {
    marginVertical: 10
  },
  marginVertical3: {
    marginVertical: 15
  },
  marginVertical4: {
    marginVertical: 20
  },

  textBold: {
    fontWeight: 'bold'
  },

  textSemiBold: {
    fontWeight: '700'
  }
});

export default sharedStyles;
