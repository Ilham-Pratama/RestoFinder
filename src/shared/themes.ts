import { Theme } from '@react-navigation/native';
import { DarkTheme } from '@react-navigation/native';

export const AppLightTheme: Theme = {
  dark: false,
  colors: {
    primary: '#c3f229',
    background: '#F2F2F2',
    card: '#FFF',
    text: '#1C1C1E',
    border: '#E2FFA7',
    notification: '#C1FC4C'
  }
};

export const AppDarkTheme: Theme = {
  dark: true,
  colors: {
    ...DarkTheme.colors,
    primary: '#3D3D3D',
    card: '#1A1A1A',
    background: '#121212'
  }
};
