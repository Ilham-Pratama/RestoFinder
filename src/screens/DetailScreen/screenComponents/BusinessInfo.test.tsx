import React from 'react';
import { fireEvent, render } from '@testing-library/react-native';

import { BUSINESS_DUMMY } from 'shared/mocks';
import BusinessInfo from './BusinessInfo';

describe('BusinessInfo', () => {
  it('renders BusinessInfo', () => {
    const yelpOpenUrlFn = jest.fn();
    const component = render(
      <BusinessInfo {...BUSINESS_DUMMY} onVisitYelpSite={yelpOpenUrlFn} />
    );

    expect(component.toJSON()).toMatchSnapshot();
    expect(component.getByRole('header').props).toHaveProperty(
      'children',
      'Business Info'
    );
    expect(
      component.getByText(BUSINESS_DUMMY.location.address1).props
    ).toHaveProperty('accessibilityRole', 'text');
    expect(
      component.getByText(BUSINESS_DUMMY.display_phone).props
    ).toHaveProperty('accessibilityRole', 'text');

    fireEvent.press(component.getByRole('button'));

    expect(yelpOpenUrlFn).toHaveBeenCalledWith(BUSINESS_DUMMY.url);
  });
});
