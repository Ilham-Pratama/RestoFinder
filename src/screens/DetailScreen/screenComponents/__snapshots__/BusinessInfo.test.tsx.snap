// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BusinessInfo renders BusinessInfo 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "rgb(255, 255, 255)",
      },
      Object {
        "paddingVertical": 20,
      },
      Object {
        "marginTop": 10,
      },
    ]
  }
>
  <Text
    accessibilityRole="header"
    style={
      Array [
        Array [
          Object {
            "fontSize": 22,
            "fontWeight": "700",
          },
          Object {
            "marginBottom": 10,
          },
          Object {
            "marginHorizontal": 15,
          },
        ],
        Object {
          "color": "rgb(28, 28, 30)",
        },
      ]
    }
  >
    Business Info
  </Text>
  <View
    style={
      Array [
        Object {
          "borderBottomColor": "#e0e0e0",
        },
        Object {
          "alignItems": "center",
          "borderBottomWidth": 1,
          "flexDirection": "row",
          "justifyContent": "space-between",
        },
        Object {
          "paddingVertical": 20,
        },
        Object {
          "marginHorizontal": 15,
        },
      ]
    }
  >
    <View>
      <Text
        accessibilityRole="text"
        style={
          Array [
            undefined,
            Object {
              "color": "rgb(28, 28, 30)",
            },
          ]
        }
      >
        800 N Point St
      </Text>
      <Text
        accessibilityRole="text"
        style={
          Array [
            undefined,
            Object {
              "color": "rgb(28, 28, 30)",
            },
          ]
        }
      >
        San Francisco
        , 
        CA
         
        94109
      </Text>
    </View>
    <RNSVGSvgView
      accessibilityRole="image"
      align="xMidYMid"
      bbHeight={20}
      bbWidth={20}
      fill="rgb(28, 28, 30)"
      focusable={false}
      height={20}
      meetOrSlice={0}
      minX={0}
      minY={0}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          Object {
            "flex": 0,
            "height": 20,
            "width": 20,
          },
        ]
      }
      vbHeight={512}
      vbWidth={512}
      width={20}
    >
      <RNSVGGroup
        fill={4280032286}
        propList={
          Array [
            "fill",
          ]
        }
      >
        <RNSVGPath
          d="M497.1 222.1l-208.1-208.1c-9.364-9.364-21.62-14.04-33.89-14.03C243.7 .0092 231.5 4.686 222.1 14.03L14.03 222.1C4.676 231.5 .0002 243.7 .0004 255.1c.0002 12.26 4.676 24.52 14.03 33.87l208.1 208.1C231.5 507.3 243.7 511.1 256 511.1c12.26 0 24.52-4.677 33.87-14.03l208.1-208.1c9.352-9.353 14.03-21.61 14.03-33.87C511.1 243.7 507.3 231.5 497.1 222.1zM410.5 252l-96 84c-10.79 9.545-26.53 .9824-26.53-12.03V272H223.1l-.0001 48C223.1 337.6 209.6 352 191.1 352S159.1 337.6 159.1 320V240c0-17.6 14.4-32 32-32h95.1V156c0-13.85 16.39-20.99 26.53-12.03l96 84C414 231 415.1 235.4 415.1 240S414 249 410.5 252z"
        />
      </RNSVGGroup>
    </RNSVGSvgView>
  </View>
  <View
    style={
      Array [
        Object {
          "borderBottomColor": "#e0e0e0",
        },
        Object {
          "alignItems": "center",
          "borderBottomWidth": 1,
          "flexDirection": "row",
          "justifyContent": "space-between",
        },
        Object {
          "paddingVertical": 20,
        },
        Object {
          "marginHorizontal": 15,
        },
      ]
    }
  >
    <Text
      accessibilityRole="text"
      style={
        Array [
          undefined,
          Object {
            "color": "rgb(28, 28, 30)",
          },
        ]
      }
    >
      (415) 749-2060
    </Text>
    <RNSVGSvgView
      accessibilityRole="image"
      align="xMidYMid"
      bbHeight={20}
      bbWidth={20}
      fill="rgb(28, 28, 30)"
      focusable={false}
      height={20}
      meetOrSlice={0}
      minX={0}
      minY={0}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          Object {
            "flex": 0,
            "height": 20,
            "width": 20,
          },
        ]
      }
      vbHeight={512}
      vbWidth={512}
      width={20}
    >
      <RNSVGGroup
        fill={4280032286}
        propList={
          Array [
            "fill",
          ]
        }
      >
        <RNSVGPath
          d="M511.2 387l-23.25 100.8c-3.266 14.25-15.79 24.22-30.46 24.22C205.2 512 0 306.8 0 54.5c0-14.66 9.969-27.2 24.22-30.45l100.8-23.25C139.7-2.602 154.7 5.018 160.8 18.92l46.52 108.5c5.438 12.78 1.77 27.67-8.98 36.45L144.5 207.1c33.98 69.22 90.26 125.5 159.5 159.5l44.08-53.8c8.688-10.78 23.69-14.51 36.47-8.975l108.5 46.51C506.1 357.2 514.6 372.4 511.2 387z"
        />
      </RNSVGGroup>
    </RNSVGSvgView>
  </View>
  <View
    accessibilityRole="button"
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "alignItems": "center",
          "borderBottomWidth": 1,
          "flexDirection": "row",
          "justifyContent": "space-between",
        },
        Object {
          "paddingVertical": 20,
        },
        Object {
          "paddingHorizontal": 15,
        },
        Object {
          "borderBottomWidth": 0,
        },
      ]
    }
  >
    <Text
      style={
        Array [
          undefined,
          Object {
            "color": "rgb(28, 28, 30)",
          },
        ]
      }
    >
      Visit Yelp Site
    </Text>
    <RNSVGSvgView
      accessibilityRole="image"
      align="xMidYMid"
      bbHeight={20}
      bbWidth={20}
      fill="rgb(28, 28, 30)"
      focusable={false}
      height={20}
      meetOrSlice={0}
      minX={0}
      minY={0}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          Object {
            "flex": 0,
            "height": 20,
            "width": 20,
          },
        ]
      }
      vbHeight={512}
      vbWidth={448}
      width={20}
    >
      <RNSVGGroup
        fill={4280032286}
        propList={
          Array [
            "fill",
          ]
        }
      >
        <RNSVGPath
          d="M256 64C256 46.33 270.3 32 288 32H415.1C415.1 32 415.1 32 415.1 32C420.3 32 424.5 32.86 428.2 34.43C431.1 35.98 435.5 38.27 438.6 41.3C438.6 41.35 438.6 41.4 438.7 41.44C444.9 47.66 447.1 55.78 448 63.9C448 63.94 448 63.97 448 64V192C448 209.7 433.7 224 416 224C398.3 224 384 209.7 384 192V141.3L214.6 310.6C202.1 323.1 181.9 323.1 169.4 310.6C156.9 298.1 156.9 277.9 169.4 265.4L338.7 96H288C270.3 96 256 81.67 256 64V64zM0 128C0 92.65 28.65 64 64 64H160C177.7 64 192 78.33 192 96C192 113.7 177.7 128 160 128H64V416H352V320C352 302.3 366.3 288 384 288C401.7 288 416 302.3 416 320V416C416 451.3 387.3 480 352 480H64C28.65 480 0 451.3 0 416V128z"
        />
      </RNSVGGroup>
    </RNSVGSvgView>
  </View>
</View>
`;
